tests/NoMissingTypeExposeTest.elm

```diff
@@ -457,6 +457,228 @@ type Mood
     = Happy Happiness
 
 
+type Happiness
+    = Ecstatic
+    | FineIGuess
+    | Unhappy
+"""
+                    ]
+    , test "reports when an exposed typed type alias uses a private type" <|
+        \() ->
+            """
+module Happiness exposing (Mood)
+
+
+type alias Mood =
+    Maybe Happiness
+
+
+type Happiness
+    = Ecstatic
+    | FineIGuess
+    | Unhappy
+"""
+                |> Review.Test.run rule
+                |> Review.Test.expectErrors
+                    [ Review.Test.error
+                        { message = "Private type `Happiness` used by exposed function"
+                        , details =
+                            [ "Type `Happiness` is not exposed but is used by an exposed function."
+                            , "Callers of this function will not be able to annotate other functions or variables that use this type outside of the module. You should expose this type or an alias of this type."
```

Comment:
I don't know what to suggest here, but "Callers of this function" is not right.

Reply:
Discussed on slack and changed to:
> Users of this module will not be able to annotate a value of this type if they wanted to. You should expose this type or an alias of this type.